docs/source/preparing_datasets/basic_dataset_conversion.md

```diff
@@ -0,0 +1,330 @@
+| Numpy Signed Int   | 'int64'       | `Int64`      | uses  `numpy.int64`      |
+| Numpy Float        | 'float16'     | `Float16`    | uses  `numpy.float16`    |
+| Numpy Float        | 'float32'     | `Float32`    | uses  `numpy.float32`    |
+| Numpy Float        | 'float64'     | `Float64`    | uses  `numpy.float64`    |
+| Numerical String   | 'str_int'     | `StrInt`     | stores in UTF-8          |
+| Numerical String   | 'str_float'   | `StrFloat`   | stores in UTF-8          |
+| Numerical String   | 'str_decimal' | `StrDecimal` | stores in UTF-8          |
+| Image              | 'pil'         | `PIL`        | raw PIL image            |
+| Image              | 'jpeg'        | `JPEG`       | PIL image as JPEG        |
+| Image              | 'png'         | `PNG`        | PIL image as PNG         |
+| Pickle             | 'pkl'         | `Pickle`     | arbitrary Python objects |
+| JSON               | 'json'        | `JSON`       | arbitrary data as JSON   |
+
+Here's an example where the field `x` is an image, and `y` is a class label, as an integer.
+<!--pytest.mark.skip-->
+```python
+column = {
+    'x': 'jpeg',
+    'y': 'int8',
+}
+```
+
+If the data type you need is not listed in the above table, then you can write your own data type class with `encode` and `decode` methods in it and patch it inside streaming. For example, let's say, you wanted to add a `complex128` data type (64 bits each for real and imaginary parts):
+
+<!--pytest.mark.skip-->
+```python
+import numpy as np
+from typing import Any
+
+from streaming.base.format.mds.encodings import Encoding, _encodings
+
+class Complex128(Encoding):
+
+    def encode(self, obj: Any) -> bytes:
+        return np.complex128(obj).tobytes()
+
+    def decode(self, data: bytes) -> Any:
+        return np.frombuffer(data, np.complex128)[0]
+
+_encodings['complex128'] = Complex128
+```
+
+4. An optional shard `size_limit`, in bytes, for each *uncompressed* shard file. This defaults to 67 MB. Specify this as a number of bytes, either directly as an `int`, or a human-readable suffix:
+
+<!--pytest.mark.skip-->
+```python
+size_limit = 1024 # 1kB limit, as an int
+size_limit = "1kb" # 1kB limit, as a human-readable string
+```
+Shard file size depends on the dataset size, but generally, too small of a shard size creates a ton of shard files and heavy network overheads, and too large of a shard size creates fewer shard files, but downloads are less balanced. A shard size of between 50-100MB works well in practice.
+
+5. An optional `compression` algorithm name (and level) if you would like to compress the shard files. This can reduce egress costs during training. StreamingDataset will uncompress shard files upon download during training. You can control whether to keep compressed shard files locally during training with the `keep_zip` flag -- more information [here](../dataset_configuration/shard_retrieval.md#Keeping-compressed-shards).
+
+Supported compression algorithms:
```

Comment:
there is perfectly fine code in the repo to vis the different compression algos on (1) enc time, (2) enc size, and (3) dec time, and we should run the numbers on different kinds of shards and include the plots here because results are what people care about

same for hashing iirc

need to note that the levels of different algos, when they exist, do not map in any uniform way to each other

Reply:
> need to note that the levels of different algos, when they exist, do not map in any uniform way to each other

@knighton could you clarify this please? And add a suggestion if possible?

As for the visualization, that can go in a different PR as well, but yes that would be nice to add.